453c670ab8f46082056f6c3b7aa65d846961c5a1..8e537aa709e45bd6d238ae41814721068ad4d626

From da42cf905d07613ad5243ddee16486369025eb6e Mon Sep 17 00:00:00 2001
From: Henri Verbeet <hverbeet@codeweavers.com>
Date: Wed, 8 May 2019 17:51:36 +0430
Subject: [PATCH] wined3d: Store the glEnableWINE and glDisableWINE pointers in
 the wined3d_gl_info structure.

Signed-off-by: Henri Verbeet <hverbeet@codeweavers.com>
Signed-off-by: Alexandre Julliard <julliard@winehq.org>
---

 dlls/wined3d/context.c         | 2 +-
 dlls/wined3d/directx.c         | 4 ++--
 dlls/wined3d/gl_compat.c       | 8 ++++----
 dlls/wined3d/state.c           | 4 ++--
 dlls/wined3d/wined3d_gl.h      | 2 --
 dlls/wined3d/wined3d_private.h | 3 +++
 6 files changed, 12 insertions(+), 11 deletions(-)


diff --git a/dlls/wined3d/context.c b/dlls/wined3d/context.c
index 6530b6fc37e..28107d7f3c3 100644
--- a/dlls/wined3d/context.c
+++ b/dlls/wined3d/context.c
@@ -2949,5 +2949,5 @@ void context_apply_ffp_blit_state(struct wined3d_context *context, const struct
     gl_info->gl_ops.gl.p_glDisable(GL_LIGHTING);
     context_invalidate_state(context, STATE_RENDER(WINED3D_RS_LIGHTING));
-    glDisableWINE(GL_FOG);
+    gl_info->p_glDisableWINE(GL_FOG);
     context_invalidate_state(context, STATE_RENDER(WINED3D_RS_FOGENABLE));
 
diff --git a/dlls/wined3d/directx.c b/dlls/wined3d/directx.c
index f18c4ace1ac..13a8e8a5264 100644
--- a/dlls/wined3d/directx.c
+++ b/dlls/wined3d/directx.c
@@ -6622,6 +6622,6 @@ static BOOL wined3d_adapter_init(struct wined3d_adapter *adapter, UINT ordinal,
 #endif
 
-    glEnableWINE = gl_info->gl_ops.gl.p_glEnable;
-    glDisableWINE = gl_info->gl_ops.gl.p_glDisable;
+    gl_info->p_glEnableWINE = gl_info->gl_ops.gl.p_glEnable;
+    gl_info->p_glDisableWINE = gl_info->gl_ops.gl.p_glDisable;
 
     if (!AllocateLocallyUniqueId(&adapter->luid))
diff --git a/dlls/wined3d/gl_compat.c b/dlls/wined3d/gl_compat.c
index 7e6b5101bc4..a2fb09c39e1 100644
--- a/dlls/wined3d/gl_compat.c
+++ b/dlls/wined3d/gl_compat.c
@@ -438,8 +438,8 @@ void install_gl_compat_wrapper(struct wined3d_gl_info *gl_info, enum wined3d_gl_
             old_fogcoord_glFogfv = gl_info->gl_ops.gl.p_glFogfv;
             gl_info->gl_ops.gl.p_glFogfv = wine_glFogfv;
-            old_fogcoord_glEnable = glEnableWINE;
-            glEnableWINE = wine_glEnable;
-            old_fogcoord_glDisable = glDisableWINE;
-            glDisableWINE = wine_glDisable;
+            old_fogcoord_glEnable = gl_info->p_glEnableWINE;
+            gl_info->p_glEnableWINE = wine_glEnable;
+            old_fogcoord_glDisable = gl_info->p_glDisableWINE;
+            gl_info->p_glDisableWINE = wine_glDisable;
 
             old_fogcoord_glVertex4f = gl_info->gl_ops.gl.p_glVertex4f;
diff --git a/dlls/wined3d/state.c b/dlls/wined3d/state.c
index e3c39f64e0d..74ea6ffccba 100644
--- a/dlls/wined3d/state.c
+++ b/dlls/wined3d/state.c
@@ -1128,5 +1128,5 @@ void state_fog_fragpart(struct wined3d_context *context, const struct wined3d_st
     {
         /* No fog? Disable it, and we're done :-) */
-        glDisableWINE(GL_FOG);
+        gl_info->p_glDisableWINE(GL_FOG);
         checkGLcall("glDisable GL_FOG");
         return;
@@ -1265,5 +1265,5 @@ void state_fog_fragpart(struct wined3d_context *context, const struct wined3d_st
     }
 
-    glEnableWINE(GL_FOG);
+    gl_info->p_glEnableWINE(GL_FOG);
     checkGLcall("glEnable GL_FOG");
     if (new_source != context->fog_source || fogstart == fogend)
diff --git a/dlls/wined3d/wined3d_gl.h b/dlls/wined3d/wined3d_gl.h
index ed439e2a2ca..3c69c0a366f 100644
--- a/dlls/wined3d/wined3d_gl.h
+++ b/dlls/wined3d/wined3d_gl.h
@@ -29,6 +29,4 @@
 #define GL_COMPRESSED_LUMINANCE_ALPHA_3DC_ATI 0x8837  /* not in the gl spec */
 
-void (WINE_GLAPI *glDisableWINE)(GLenum cap) DECLSPEC_HIDDEN;
-void (WINE_GLAPI *glEnableWINE)(GLenum cap) DECLSPEC_HIDDEN;
 
 /* OpenGL extensions. */
diff --git a/dlls/wined3d/wined3d_private.h b/dlls/wined3d/wined3d_private.h
index 4f02ae465c1..5c50ef88af9 100644
--- a/dlls/wined3d/wined3d_private.h
+++ b/dlls/wined3d/wined3d_private.h
@@ -2582,4 +2582,7 @@ struct wined3d_gl_info
     struct wined3d_format *formats;
     unsigned int format_count;
+
+    void (WINE_GLAPI *p_glDisableWINE)(GLenum cap);
+    void (WINE_GLAPI *p_glEnableWINE)(GLenum cap);
 };
 
--
